src/course-unit/course-xblock/CourseXBlock.test.jsx

```diff
@@ -0,0 +1,108 @@
+import { render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { initializeMockApp } from '@edx/frontend-platform';
+import { AppProvider } from '@edx/frontend-platform/react';
+
+import { courseVerticalChildrenMock } from '../__mocks__';
+import CourseXBlock from './CourseXBlock';
+
+import messages from './messages';
+import deleteModalMessages from '../../generic/delete-modal/messages';
+
+let store;
+const handleDeleteMock = jest.fn();
+const handleDuplicateMock = jest.fn();
+const XblockData = courseVerticalChildrenMock.children[0];
+const unitXBlockActionsMock = {
+  handleDelete: handleDeleteMock,
+  handleDuplicate: handleDuplicateMock,
+};
+
+const renderComponent = () => render(
+  <AppProvider store={store}>
+    <IntlProvider locale="en">
+      <CourseXBlock
+        id={XblockData.block_id}
+        title={XblockData.block_id}
+        unitXBlockActions={unitXBlockActionsMock}
+        shouldScroll={false}
+      />
+    </IntlProvider>
+  </AppProvider>,
+);
+
+describe('<CourseXBlock />', () => {
+  beforeEach(async () => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: true,
+        roles: [],
+      },
+    });
+  });
+
+  it('render CourseXBlock component correctly', async () => {
+    const { getByText, getByLabelText } = renderComponent();
+
+    await waitFor(() => {
+      expect(getByText(XblockData.block_id)).toBeInTheDocument();
+      expect(getByLabelText(messages.blockAltButtonEdit.defaultMessage)).toBeInTheDocument();
+      expect(getByLabelText(messages.blockActionsDropdownAlt.defaultMessage)).toBeInTheDocument();
+    });
+  });
+
+  it('render CourseXBlock component action dropdown correctly', async () => {
+    const { getByText, getByLabelText } = renderComponent();
+
+    await waitFor(() => {
+      userEvent.click(getByLabelText(messages.blockActionsDropdownAlt.defaultMessage));
+      expect(getByText(messages.blockLabelButtonCopy.defaultMessage)).toBeInTheDocument();
```

Comment:
[question]: For me, it is preferable to search for elements by role and text. Thus, we make maximum use of RTL API. 
@ihor-romaniuk what do you think?

```suggestion
      expect(getByRole('button', { name: messages.blockLabelButtonCopy.defaultMessage })).toBeInTheDocument();
```